packages/next-on-pages/templates/_worker.js/utils/cache.ts

```diff
@@ -71,6 +70,22 @@ export async function handleSuspenseCacheRequest(request: Request) {
  * @returns Adaptor for the suspense cache.
  */
 export async function getSuspenseCacheAdaptor(): Promise<CacheAdaptor> {
-	// TODO: Try to lazy import the custom cache adaptor.
-	return new CacheApiAdaptor();
+	if (process.env.__NEXT_ON_PAGES__KV_SUSPENSE_CACHE) {
```

Comment:
I think we haven't considered the case in which the user provides a `__NEXT_ON_PAGES__KV_SUSPENSE_CACHE` binding but that is not a KV

(although it seems quite unlikely to me 🤷)

Should we show a warning if this binding is not a KV and still use the default cache API adaptor? (if it doesn't contain `get`, `set`)?
(That developers could see locally or if they tail their worker)

Reply:
if someone wants to set the wrong binding for that key, that's a user error and will probably result in an internal server error. i don't see the problem here

Reply:
> if someone wants to set the wrong binding for that key, that's a user error and will probably result in an internal server error. i don't see the problem here

I don't think people "want" to make mistakes 😅

I'm just saying that if the user makes a mistake we can either make the whole thing blow up (like it currently would) or try to gracefully handle the situation.

I would prefer to gracefully handle it, but you could argue that that would make the error less evident/discoverable, so I am fine with either.